src/nvim/ops.c

```diff
@@ -4643,15 +4654,11 @@ void write_reg_contents_lst(int name, char_u **strings, int maxlen,
 /// @param block_len width of visual block
 void write_reg_contents_ex(int name,
                            const char_u *str,
-                           ssize_t len,
+                           size_t len,
                            bool must_append,
                            int yank_type,
                            long block_len)
 {
-  if (len < 0) {
```

Comment:
it is indeed a common idiom. I don't know where I got the idea but I thought that we indeed wanted to get rid of it. Maybe @justinmk can light us on that. If I must revert, do you think it's best to revert all cases or just to handle again the `-1`?